Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/SMS.hs

```diff
@@ -51,3 +65,68 @@ sendSMS merchantId = Sms.sendSMS handler
       case merchantSmsServiceConfig.serviceConfig of
         DMSC.SmsServiceConfig msc -> pure msc
         _ -> throwError $ InternalError "Unknown Service Config"
+
+data DashboardMessageType = BOOKING | ENDRIDE | ONBOARDING | CASH_COLLECTED deriving (Show, Generic, Eq)
+
+sendDashboardSms ::
+  ( EsqDBReplicaFlow m r,
+    HasCacheConfig r,
+    HasFlowEnv m r '["smsCfg" ::: SmsConfig],
+    EsqDBFlow m r,
+    EsqLocDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r,
+    CoreMetrics m
+  ) =>
+  Id Merchant ->
+  DashboardMessageType ->
+  Maybe DR.Ride ->
+  Id DP.Person ->
+  Maybe SRB.Booking ->
+  HighPrecMoney ->
+  m ()
+sendDashboardSms merchantId messageType mbRide driverId mbBooking amount = do
+  transporterConfig <- SCT.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigNotFound merchantId.getId)
```

Comment:
run in replica

Reply:
not able to do runInReplica... wherever we are fetching the transporter config and merchant service usage config, we are not able to do runInReplica - ( Some constraint problems are arising for MonadReader) - not able to solve it